packages/runtime-dom/src/jsx.ts

```diff
@@ -406,7 +406,7 @@ export interface DataHTMLAttributes extends HTMLAttributes {
 
 export interface DetailsHTMLAttributes extends HTMLAttributes {
   open?: Booleanish
-  onToggle?: Event
+  onToggle?: (payload: MouseEvent) => void
```

Comment:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDetailsElement/toggle_event#event_type), it should be a generic `Event`.
According to [the HTML spec](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-details-element:event-toggle), though, it should be a `ToggleEvent`.

Either way, it's not a `MouseEvent`.

Reply:
Thanks for the detailed explanation, updated.